applications/CompressiblePotentialFlowApplication/python_scripts/compute_lift_process.py

```diff
@@ -14,46 +26,90 @@ def __init__(self, Model, settings ):
             "model_part_name": "please specify the model part that contains the surface nodes",
             "velocity_infinity": [1.0,0.0,0.0],
             "reference_area": 1.0,
+            "moment_reference_point" : [],
             "create_output_file": false
         }''')
 
         settings.ValidateAndAssignDefaults(default_parameters)
 
         self.body_model_part = Model[settings["model_part_name"].GetString()]
-        self.velocity_infinity = [0,0,0]
-        self.velocity_infinity[0] = settings["velocity_infinity"][0].GetDouble()
-        self.velocity_infinity[1] = settings["velocity_infinity"][1].GetDouble()
-        self.velocity_infinity[2] = settings["velocity_infinity"][2].GetDouble()
+        self.fluid_model_part = self.body_model_part.GetRootModelPart()
         self.reference_area =  settings["reference_area"].GetDouble()
         self.create_output_file = settings["create_output_file"].GetBool()
+        self.moment_reference_point = settings["moment_reference_point"].GetVector()
+        
 
     def ExecuteFinalizeSolutionStep(self):
         KratosMultiphysics.Logger.PrintInfo('ComputeLiftProcess','COMPUTE LIFT')
 
-        rx = 0.0
-        ry = 0.0
-        rz = 0.0
+        total_force = KratosMultiphysics.Vector(3)
+        total_moment = KratosMultiphysics.Vector(3)
 
         for cond in self.body_model_part.Conditions:
             n = cond.GetValue(KratosMultiphysics.NORMAL)
             cp = cond.GetValue(KratosMultiphysics.PRESSURE)
 
-            rx += n[0]*cp
-            ry += n[1]*cp
-            rz += n[2]*cp
+            # Computing forces
+            total_force += n.__mul__(cp)
+
+            # Computing moment
+            mid_point = cond.GetGeometry().Center()
+            lever = mid_point-self.moment_reference_point
+            total_moment += CrossProduct(lever, n*(-cp))
+
+        force_coefficient = total_force.__div__(self.reference_area)
```

Comment:
same

Reply:
This does not work. I get the following error:

force_coefficient = total_force/self.reference_area
TypeError: unsupported operand type(s) for /: 'Kratos.Vector' and 'float'

Reply:
ok

Reply:
Thanks for the hint!

Regarding your question: actually the problem is I again chose a bad name. The variable total_force is actually already a coefficient with units of meter because it is the result of multiplying the pressure coefficient with the normal and integrating over the airfoil. The force_coefficient is a non dimensional coefficient resulting from dividing the total force with units Newton/m by the dynamic pressure and the reference length which is typically selected as the chord length. I changed this and hopefully chose better names. Let me know if the names can still be improved.